### PR TITLE
Expose container_id in mine.get_docker

### PR DESCRIPTION
**Pull request by ticosax**
Because sometimes we need to be able to perform additional introspection.

`Note` I did it in a way that is backward compatible, but I believe the result is unpleasant to use. I think `mine.get_docker` deserve a new interface.

### CODE REVIEW

**Comment by cachedout** [comment]
Could you please fix the minor lint error? Thanks.

**Comment by ticosax** [comment]
@cachedout thank you for the notification. I pushed the fix.

File: salt/modules/mine.py [hunk 1]
```diff
@@ -294,13 +294,17 @@ def flush():
...
+def get_docker(interfaces=None, cidrs=None, with_container_id=False):
     '''
     Get all mine data for 'docker.get_containers' and run an aggregation
     routine. The "interfaces" parameter allows for specifying which network
     interfaces to select ip addresses from. The "cidrs" parameter allows for
     specifying a list of cidrs which the ip address must match.
 
+    with_container_id
+        Boolean, to expose container_id in the list of results
+
```
Comment (jfindlay) [inline salt/modules/mine.py]: @ticosax 
```
with_container_id
    Boolean, to expose container_id in the list of results

    .. versionadded:: 2015.8.2
```

**Comment by ticosax** [comment]
Thank you @jfindlay

**Comment by basepi** [comment]
Looks great, thanks!

**Comment by blueyed** [comment]
@basepi 
Just a note: `fixup! *`-commits are meant to be squashed, not merged as-is.  See e.g. https://coderwall.com/p/hh-4ea/git-rebase-autosquash.

I have a pre-commit hook to take care of that (https://github.com/blueyed/dotfiles/blob/master/config/git/template/hooks/pre-push), but it does not help when merging from the Github UI.

**Comment by basepi** [comment]
That's fine, if the user wants to squash them. But as a maintainer, it would lose authorship if I squashed those commits, wouldn't it?

**Comment by blueyed** [comment]
@basepi 
No, you can amend existing commits, without changing the Author (it is the default).
That's true also for cherry-picking etc.

But in general you'd wait for the submitter to do it and force-push the result, and then merge that.

**Comment by basepi** [comment]
/me nods

Thanks for the heads up.

**Pull request merged**